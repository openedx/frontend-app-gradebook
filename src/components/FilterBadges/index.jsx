import { connect } from 'react-redux';

import React from 'react';
import PropTypes from 'prop-types';
import initialFilters from '../../data/constants/filters';

function FilterBadge({
  name, value, onClick, showValue,
}) {
  return (
    <div>
      <span className="badge badge-info">
        <span>{name}{showValue && `: ${value}`}</span>
        <button type="button" className="btn-info" aria-label="Close" onClick={onClick}>
          <span aria-hidden="true">&times;</span>
        </button>
      </span>
      <br />
    </div>
  );
}

FilterBadge.defaultProps = {
  showValue: true,
};

FilterBadge.propTypes = {
  name: PropTypes.string.isRequired,
  value: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.bool,
  ]).isRequired,
  onClick: PropTypes.func.isRequired,
  showValue: PropTypes.bool,
};

function RangeFilterBadge({
  displayName,
  filterName1,
  filterValue1,
  filterName2,
  filterValue2,
  handleBadgeClose,
}) {
  return ((filterValue1 !== initialFilters[filterName1])
  || (filterValue2 !== initialFilters[filterName2]))
   && (
   <FilterBadge
     name={displayName}
     value={`${filterValue1} - ${filterValue2}`}
     onClick={handleBadgeClose}
   />
   );
}
RangeFilterBadge.propTypes = {
  displayName: PropTypes.string.isRequired,
  filterName1: PropTypes.string.isRequired,
  filterValue1: PropTypes.string.isRequired,
  filterName2: PropTypes.string.isRequired,
  filterValue2: PropTypes.string.isRequired,
  handleBadgeClose: PropTypes.func.isRequired,
};

function SingleValueFilterBadge({
  displayName, filterName, filterValue, handleBadgeClose, showValue,
}) {
  return (filterValue !== initialFilters[filterName])
  && (
  <FilterBadge
    name={displayName}
    value={filterValue}
    onClick={handleBadgeClose}
    showValue={showValue}
  />
  );
}

SingleValueFilterBadge.defaultProps = {
  showValue: true,
};

SingleValueFilterBadge.propTypes = {
  displayName: PropTypes.string.isRequired,
  filterName: PropTypes.string.isRequired,
  filterValue: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.bool,
  ]).isRequired,
  handleBadgeClose: PropTypes.func.isRequired,
  showValue: PropTypes.bool,
};

function FilterBadges({
  assignment,
  assignmentType,
  track,
  cohort,
  assignmentGradeMin,
  assignmentGradeMax,
  courseGradeMin,
  courseGradeMax,
  includeCourseRoleMembers,
  handleFilterBadgeClose,
}) {
  return (
    <div>
      <SingleValueFilterBadge
        displayName="Assignment Type"
        filterName="assignmentType"
        filterValue={assignmentType}
        handleBadgeClose={handleFilterBadgeClose(['assignmentType'])}
      />
      <SingleValueFilterBadge
        displayName="Assignment"
        filterName="assignment"
        filterValue={assignment}
        handleBadgeClose={handleFilterBadgeClose(['assignment', 'assignmentGradeMax', 'assignmentGradeMin'])}
      />
      <RangeFilterBadge
        displayName="Assignment Grade"
        filterName1="assignmentGradeMin"
        filterValue1={assignmentGradeMin}
        filterName2="assignmentGradeMax"
        filterValue2={assignmentGradeMax}
        handleBadgeClose={handleFilterBadgeClose(['assignmentGradeMin', 'assignmentGradeMax'])}
      />
      <RangeFilterBadge
        displayName="Course Grade"
        filterName1="courseGradeMin"
        filterValue1={courseGradeMin}
        filterName2="courseGradeMax"
        filterValue2={courseGradeMax}
        handleBadgeClose={handleFilterBadgeClose(['courseGradeMin', 'courseGradeMax'])}
      />
      <SingleValueFilterBadge
        displayName="Track"
        filterName="track"
        filterValue={track}
        handleBadgeClose={handleFilterBadgeClose(['track'])}
      />
      <SingleValueFilterBadge
        displayName="Cohort"
        filterName="cohort"
        filterValue={cohort}
        handleBadgeClose={handleFilterBadgeClose(['cohort'])}
      />
      <SingleValueFilterBadge
        displayName="Including Course Team Members"
        filterName="includeCourseRoleMembers"
        filterValue={includeCourseRoleMembers}
        showValue={false}
        handleBadgeClose={handleFilterBadgeClose(['includeCourseRoleMembers'])}
      />
    </div>
  );
}

const mapStateToProps = state => (
  {
    assignment: (state.filters.assignment || {}).label,
    assignmentType: state.filters.assignmentType,
    track: state.filters.track,
    cohort: state.filters.cohort,
    assignmentGradeMin: state.filters.assignmentGradeMin,
    assignmentGradeMax: state.filters.assignmentGradeMax,
    courseGradeMin: state.filters.courseGradeMin,
    courseGradeMax: state.filters.courseGradeMax,
    includeCourseRoleMembers: state.filters.includeCourseRoleMembers,
  }
);

const ConnectedFilterBadges = connect(mapStateToProps)(FilterBadges);
export default ConnectedFilterBadges;

FilterBadges.defaultProps = {
  assignment: initialFilters.assignmentType,
  assignmentType: initialFilters.assignmentType,
  track: initialFilters.track,
  cohort: initialFilters.cohort,
  assignmentGradeMin: initialFilters.assignmentGradeMin,
  assignmentGradeMax: initialFilters.assignmentGradeMax,
  courseGradeMin: initialFilters.courseGradeMin,
  courseGradeMax: initialFilters.courseGradeMax,
  includeCourseRoleMembers: initialFilters.includeCourseRoleMembers,
};

FilterBadges.propTypes = {
  assignment: PropTypes.string,
  assignmentType: PropTypes.string,
  track: PropTypes.string,
  cohort: PropTypes.string,
  assignmentGradeMin: PropTypes.string,
  assignmentGradeMax: PropTypes.string,
  courseGradeMin: PropTypes.string,
  courseGradeMax: PropTypes.string,
  includeCourseRoleMembers: PropTypes.bool,
  handleFilterBadgeClose: PropTypes.func.isRequired,
};
