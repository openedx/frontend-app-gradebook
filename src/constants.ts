export const appId = 'org.openedx.frontend.app.gradebook';
