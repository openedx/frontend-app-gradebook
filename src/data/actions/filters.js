import filterSelectors from 'data/selectors/filters';
import initialFilters from '../constants/filters';
import {
  INITIALIZE_FILTERS,
  RESET_FILTERS,
  UPDATE_ASSIGNMENT_FILTER,
  UPDATE_ASSIGNMENT_LIMITS,
  UPDATE_COURSE_GRADE_LIMITS,
  UPDATE_INCLUDE_COURSE_ROLE_MEMBERS,
} from '../constants/actionTypes/filters';
import { fetchGrades } from './grades';

const { allFilters } = filterSelectors;

const initializeFilters = ({
  assignment = initialFilters.assignment,
  assignmentType = initialFilters.assignmentType,
  track = initialFilters.track,
  cohort = initialFilters.cohort,
  assignmentGradeMin = initialFilters.assignmentGradeMin,
  assignmentGradeMax = initialFilters.assignmentGradeMax,
  courseGradeMin = initialFilters.courseGradeMin,
  courseGradeMax = initialFilters.assignmentGradeMax,
  includeCourseRoleMembers = initialFilters.includeCourseRoleMembers,
}) => ({
  type: INITIALIZE_FILTERS,
  data: {
    assignment: { id: assignment },
    assignmentType,
    track,
    cohort,
    assignmentGradeMin,
    assignmentGradeMax,
    courseGradeMin,
    courseGradeMax,
    includeCourseRoleMembers: Boolean(includeCourseRoleMembers),
  },
});

const resetFilters = filterNames => ({
  type: RESET_FILTERS,
  filterNames,
});

const updateAssignmentFilter = assignment => ({
  type: UPDATE_ASSIGNMENT_FILTER,
  data: assignment,
});

const updateAssignmentLimits = (minGrade, maxGrade) => ({
  type: UPDATE_ASSIGNMENT_LIMITS,
  data: { minGrade, maxGrade },
});

const updateCourseGradeFilter = (courseGradeMin, courseGradeMax, courseId) => ({
  type: UPDATE_COURSE_GRADE_LIMITS,
  data: {
    courseGradeMin,
    courseGradeMax,
    courseId,
  },
});

const updateIncludeCourseRoleMembersFilter = (includeCourseRoleMembers) => ({
  type: UPDATE_INCLUDE_COURSE_ROLE_MEMBERS,
  data: {
    includeCourseRoleMembers,
  },
});

const updateIncludeCourseRoleMembers = includeCourseRoleMembers => (dispatch, getState) => {
  dispatch(updateIncludeCourseRoleMembersFilter(includeCourseRoleMembers));
  const state = getState();
  const { cohort, track, assignmentType } = allFilters(state);
  dispatch(fetchGrades(state.grades.courseId, cohort, track, assignmentType));
};

export {
  initializeFilters, resetFilters, updateAssignmentFilter,
  updateAssignmentLimits, updateCourseGradeFilter, updateIncludeCourseRoleMembers,
};
