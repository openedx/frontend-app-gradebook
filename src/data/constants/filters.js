import { StrictDict } from 'utils';

import messages from './filters.messages';

export const filters = StrictDict({
  assignment: 'assignment',
  assignmentGrade: 'assignmentGrade',
  assignmentGradeMax: 'assignmentGradeMax',
  assignmentGradeMin: 'assignmentGradeMin',
  assignmentType: 'assignmentType',
  cohort: 'cohort',
  courseGrade: 'courseGrade',
  courseGradeMax: 'courseGradeMax',
  courseGradeMin: 'courseGradeMin',
  excludedCourseRoles: 'excludedCourseRoles',
  includeCourseRoleMembers: 'includeCourseRoleMembers',
  track: 'track',
});

const initialFilters = {
  [filters.assignment]: '',
  [filters.assignmentGradeMax]: '100',
  [filters.assignmentGradeMin]: '0',
  [filters.assignmentType]: '',
  [filters.cohort]: '',
  [filters.courseGradeMax]: '100',
  [filters.courseGradeMin]: '0',
  [filters.includeCourseRoleMembers]: false,
  [filters.track]: '',
};

export const filterConfig = StrictDict({
  [filters.assignment]: {
    displayName: messages[filters.assignment],
    connectedFilters: ['assignment', 'assignmentGradeMax', 'assignmentGradeMax'],
  },
  [filters.assignmentType]: {
    displayName: messages[filters.assignmentType],
    connectedFilters: ['assignmentType'],
  },
  [filters.assignmentGrade]: {
    displayName: messages[filters.assignmentGrade],
    filterOrder: ['assignmentGradeMin', 'assignmentGradeMax'],
    connectedFilters: ['assignmentGradeMax', 'assignmentGradeMin'],
  },
  [filters.cohort]: {
    displayName: messages[filters.cohort],
    connectedFilters: ['cohort'],
  },
  [filters.courseGrade]: {
    displayName: messages[filters.courseGrade],
    filterOrder: ['courseGradeMin', 'courseGradeMax'],
    connectedFilters: ['courseGradeMax', 'courseGradeMin'],
  },
  [filters.includeCourseRoleMembers]: {
    displayName: messages[filters.includeCourseRoleMembers],
    connectedFilters: ['includeCourseRoleMembers'],
    hideValue: true,
  },
  [filters.track]: {
    displayName: messages[filters.track],
    connectedFilters: ['track'],
  },
});

export const badgeOrder = [
  filters.assignmentType,
  filters.assignment,
  filters.assignmentGrade,
  filters.courseGrade,
  filters.track,
  filters.cohort,
  filters.includeCourseRoleMembers,
];

export default initialFilters;
