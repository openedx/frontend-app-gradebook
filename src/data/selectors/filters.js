/* eslint-disable import/no-self-import */
import { StrictDict } from 'utils';

import initialFilters from 'data/constants/filters';
import simpleSelectorFactory from '../utils';
import * as module from './filters';

// Transformers
/**
 * isDefault(name, value)
 * returns true iff the value is equal to the initial filter value
 * associated with the given name
 * @param {string} name - api filter name
 * @param {string/number/bool} value - filter value
 * @return {bool} - is this the default value for the given filter?
 */
export const isDefault = (name, value) => (
  value === initialFilters[name]
);

/**
 * chooseRelevantAssignmentData(assignment)
 * formats the assignment api data for an assignment object for consumption
 * @param {object} assignment - assignment data to prepare
 * @return {object} - formatted data ({ label, subsectionLabel, type, id })
 */
export const chooseRelevantAssignmentData = ({
  label,
  subsection_name: subsectionLabel,
  category: type,
  module_id: id,
}) => ({
  label, subsectionLabel, type, id,
});

/**
 * getAssignmentsFromResultsSubstate(results)
 * returns the section_breakdown of the first results entry
 * defaulting to an empty list.
 * @param {[object[]]} results - list of result entries from grades fetch
 * @return {object} - section_breakdown of first grade entry
 */
export const getAssignmentsFromResultsSubstate = (results) => (
  (results[0] || {}).section_breakdown || []
);

/**
 * relevantAssignmentDataFromResults
 * returns assignment info from grades results for the assignment with the given id
 * @param {object} grades - grades fetch result
 * @param {string} id - selected assignment id from assignment filter
 * @return {object} assignment data with type, label, and subsectionLabel
 */
export const relevantAssignmentDataFromResults = (grades, id) => (
  module.getAssignmentsFromResultsSubstate(grades)
    .map(module.chooseRelevantAssignmentData)
    .find(assignment => assignment.id === id)
);

// Selectors
/**
 * allFilters(state)
 * returns the top-level filter state.
 * @param {object} state - redux state
 * @return {object} - filter substate from redux state
 */
export const allFilters = (state) => state.filters || {};

/**
 * areAssignmentGradeFiltersSet(state)
 * Returns true iff either assignmentGradeMax or assignmentGradeMax is set.
 * @param {object} state - redux state
 * @return {bool} - are assignmentGrade filters set?
 */
export const areAssignmentGradeFiltersSet = ({ filters }) => !!(
  filters.assignmentGradeMax || filters.assignmentGradeMin
);

/**
 * selectableAssignments(state)
 * @param {object} state - redux state
 * @return {object[]} - list of selectable assignment objects, filtered if there is an
 *   assignmentType filter selected.
 */
export const selectableAssignments = (state) => {
  const selectedAssignmentType = allFilters(state).assignmentType;
  const needToFilter = selectedAssignmentType && selectedAssignmentType !== 'All';
  const allAssignments = getAssignmentsFromResultsSubstate(state.grades.results);
  if (needToFilter) {
    return allAssignments.filter(
      (assignment) => assignment.category === selectedAssignmentType,
    );
  }
  return allAssignments;
};

/**
 * Returns the relevant assignment data for all selectable assignments
 * @param {object} state - redux state
 * @return {object[]} - object of assignment data entries [({ label, subsectionLabel, type, id })]
 */
export const selectableAssignmentLabels = (state) => (
  selectableAssignments(state).map(chooseRelevantAssignmentData)
);

export const simpleSelectors = simpleSelectorFactory(
  ({ filters }) => filters,
  [
    'assignment',
    'assignmentGradeMax',
    'assignmentGradeMin',
    'assignmentType',
    'cohort',
    'courseGradeMax',
    'courseGradeMin',
    'track',
    'includeCourseRoleMembers',
  ],
);
/**
 * Returns the id of the selected assignment filter
 * @param {object} state - redux state
 * @return {string} - assignment id
 */
export const selectedAssignmentId = (state) => (simpleSelectors.assignment(state) || {}).id;

/**
 * selectedAssignmentLabel(state)
 * Returns the label of the selected assignment filter
 * @param {object} state - redux state
 * @return {string} - assignment label
 */
export const selectedAssignmentLabel = (state) => (simpleSelectors.assignment(state) || {}).label;

/**
 * Returns the api value for excludedCourseRoles based on the
 * internal Bool value for includeCourseRoleMembers.
 * @param {object} state - redux state
 * @return {string} - '' if to be included, else 'all'
 */
export const excludedCourseRoles = (state) => (
  simpleSelectors.includeCourseRoleMembers(state) ? '' : 'all'
);

export default StrictDict({
  ...simpleSelectors,
  isDefault,
  relevantAssignmentDataFromResults,
  selectedAssignmentId,
  selectedAssignmentLabel,
  selectableAssignmentLabels,
  selectableAssignments,
  allFilters,
  areAssignmentGradeFiltersSet,
  chooseRelevantAssignmentData,
  excludedCourseRoles,
  getAssignmentsFromResultsSubstate,
});
