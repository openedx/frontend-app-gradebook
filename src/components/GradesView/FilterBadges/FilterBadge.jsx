import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { Button } from '@edx/paragon';
import { FormattedMessage } from '@edx/frontend-platform/i18n';

import selectors from 'data/selectors';

/**
 * FilterBadge
 * Base filter badge component, that displays a name and a close button.
 * If showValue is true, it will also display the included value.
 * @param {func} handleClose - close/dismiss filter event, taking a list of filternames
 *   to reset when the filter badge closes.
 * @param {string} filterName - api filter name (for redux connector)
 */
export const FilterBadge = ({
  config: {
    displayName,
    isDefault,
    hideValue,
    value,
    connectedFilters,
  },
  handleClose,
}) => !isDefault && (
  <div>
    <span className="badge badge-info">
      <span>
        <FormattedMessage {...displayName} />
      </span>
      <span>
        {!hideValue ? `: ${value}` : ''}
      </span>
      <Button
        className="btn-info"
        aria-label="close"
        onClick={handleClose(connectedFilters)}
      >
        <span aria-hidden="true">&times;</span>
      </Button>
    </span>
    <br />
  </div>
);

FilterBadge.propTypes = {
  handleClose: PropTypes.func.isRequired,
  // eslint-disable-next-line
  filterName: PropTypes.string.isRequired,
  // redux
  config: PropTypes.shape({
    connectedFilters: PropTypes.arrayOf(PropTypes.string),
    displayName: PropTypes.shape({
      defaultMessage: PropTypes.string,
    }).isRequired,
    isDefault: PropTypes.bool.isRequired,
    hideValue: PropTypes.bool,
    value: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.bool,
    ]),
  }).isRequired,
};

export const mapStateToProps = (state, ownProps) => ({
  config: selectors.root.filterBadgeConfig(state, ownProps.filterName),
});

export default connect(mapStateToProps)(FilterBadge);
