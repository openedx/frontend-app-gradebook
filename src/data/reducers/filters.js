import selectors from 'data/selectors';
import actions from '../actions/filters';
import gradeActions from '../actions/grades';
import initialFilters from '../constants/filters';

const initialState = {};

const reducer = (state = initialState, { type: actionType, payload }) => {
  switch (actionType) {
    case actions.initialize.toString():
      return {
        ...state,
        ...payload,
      };
    case actions.reset.toString(): {
      const result = { ...state };
      payload.forEach((filterName) => {
        result[filterName] = initialFilters[filterName];
      });
      return result;
    }
    case actions.update.assignment.toString():
      return {
        ...state,
        assignment: payload,
      };
    case actions.update.assignmentLimits.toString():
      return {
        ...state,
        assignmentGradeMax: payload.assignmentGradeMax,
        assignmentGradeMin: payload.assignmentGradeMin,
      };
    case actions.update.assignmentType.toString(): {
      const newState = { ...state, assignmentType: payload };
      if (payload !== '' && state.assignment && payload !== state.assignment.type) {
        newState.assignment = '';
      }
      return newState;
    }
    case actions.update.cohort.toString():
      return { ...state, cohort: payload };
    case actions.update.courseGradeLimits.toString():
      return {
        ...state,
        courseGradeMax: payload.courseGradeMax,
        courseGradeMin: payload.courseGradeMin,
      };
    case actions.update.includeCourseRoleMembers.toString():
      return { ...state, includeCourseRoleMembers: payload };
    case actions.update.track.toString():
      return { ...state, track: payload };
    case gradeActions.fetching.received.toString(): {
      const { assignment } = state;
      const { id, type } = assignment || {};
      if (id && !type) {
        const { relevantAssignmentDataFromResults } = selectors.filters;
        const relevantAssignment = relevantAssignmentDataFromResults(
          payload.grades,
          id,
        );
        return {
          ...state,
          track: payload.track,
          cohort: payload.cohort,
          assignment: relevantAssignment,
        };
      }
      return {
        ...state,
        track: payload.track,
        cohort: payload.cohort,
      };
    }
    default:
      return state;
  }
};

export { initialState };
export default reducer;
