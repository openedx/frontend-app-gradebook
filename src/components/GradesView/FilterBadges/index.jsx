/* eslint-disable import/no-named-as-default */
import React from 'react';
import PropTypes from 'prop-types';

import { badgeOrder } from 'data/constants/filters';

import FilterBadge from './FilterBadge';

/**
 * FilterBadges
 * Displays a FilterBadge for each filter type in the data model with their current values.
 * @param {func} handleClose - event taking a list of filternames to reset
 */
export const FilterBadges = ({ handleClose }) => (
  <div>
    {badgeOrder.map(filterName => (
      <FilterBadge key={filterName} {...{ handleClose, filterName }} />
    ))}
  </div>
);
FilterBadges.propTypes = {
  handleClose: PropTypes.func.isRequired,
};

export default FilterBadges;
