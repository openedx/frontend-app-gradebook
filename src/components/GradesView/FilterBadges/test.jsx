/* eslint-disable import/no-named-as-default */
import React from 'react';
import { shallow } from 'enzyme';

import * as constants from 'data/constants/filters';
import FilterBadges from '.';
import FilterBadge from './FilterBadge';

jest.mock('./FilterBadge', () => 'FilterBadge');

describe('FilterBadges', () => {
  describe('component', () => {
    let el;
    let handleClose;
    const order = ['filter1', 'filter2', 'filter3'];
    beforeEach(() => {
      handleClose = jest.fn().mockName('this.props.handleClose');
      constants.badgeOrder = order;
      el = shallow(<FilterBadges handleClose={handleClose} />);
    });
    test('snapshot - has a filterbadge with handleClose for each filter in badgeOrder', () => {
      expect(el).toMatchSnapshot();
    });
    test('has a filterbadge with handleClose for each filter in badgeOrder', () => {
      const badgeProps = el.find(FilterBadge).map(badgeEl => badgeEl.props());
      // key prop is not rendered by react
      expect(badgeProps[0]).toEqual({ filterName: order[0], handleClose });
      expect(badgeProps[1]).toEqual({ filterName: order[1], handleClose });
      expect(badgeProps[2]).toEqual({ filterName: order[2], handleClose });
      expect(badgeProps.length).toEqual(3);
    });
  });
});
