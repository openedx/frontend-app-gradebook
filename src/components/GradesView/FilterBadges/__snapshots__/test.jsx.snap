// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterBadges component snapshot - has a filterbadge with handleClose for each filter in badgeOrder 1`] = `
<div>
  <FilterBadge
    filterName="filter1"
    handleClose={[MockFunction this.props.handleClose]}
    key="filter1"
  />
  <FilterBadge
    filterName="filter2"
    handleClose={[MockFunction this.props.handleClose]}
    key="filter2"
  />
  <FilterBadge
    filterName="filter3"
    handleClose={[MockFunction this.props.handleClose]}
    key="filter3"
  />
</div>
`;
